packages/prerender-fargate/lib/prerender-fargate.ts

```diff
@@ -9,84 +9,105 @@ import * as ecrAssets from "aws-cdk-lib/aws-ecr-assets";
+ *          "https://example.com",
+ *          "https://acme.example.com"
+ *        ],
+ *        token2: [
+ *          "https://example1.com",
+ *          "https://acme.example1.com"
+ *        ]
+ *    },
+ *    ssmPathPrefix: "/prerender/recache/tokens"
+ * }
+ * });
+ * ```
+ *
+ * @param {Construct} scope The scope of the construct.
+ * @param {string} id The ID of the construct.
+ * @param {PrerenderFargateOptions} props The properties of the Prerender Fargate service.
+ *
+ * @returns {PrerenderFargate} An instance of the PrerenderFargate class.
  */
-export interface PrerenderOptions {
-  /**
-   * The name of the Prerender service.
-   */
-  prerenderName: string;
-  /**
-   * The domain name to prerender.
-   */
-  domainName: string;
-  /**
-   * The ID of the VPC to deploy the Fargate service in.
-   */
-  vpcId?: string;
-  /**
-   * The name of the S3 bucket to store prerendered pages in.
-   */
-  bucketName?: string;
-  /**
-   * The number of days to keep prerendered pages in the S3 bucket before expiring them.
-   */
-  expirationDays?: number;
-  /**
-   * A list of tokens to use for authentication with the Prerender service.
-   */
-  tokenList: Array<string>;
-  /**
-   * The ARN of the SSL certificate to use for HTTPS connections.
-   */
-  certificateArn: string;
-  /**
-   * The desired number of Fargate instances to run.
-   */
-  desiredInstanceCount?: number;
-  /**
-   * The maximum number of Fargate instances to run.
-   */
-  maxInstanceCount?: number;
-  /**
-   * The amount of CPU to allocate to each Fargate instance.
-   */
-  instanceCPU?: number;
-  /**
-   * The amount of memory to allocate to each Fargate instance.
-   */
-  instanceMemory?: number;
-  /**
-   * Whether to enable caching of HTTP redirects.
-   */
-  enableRedirectCache?: string;
-  /**
-   * Whether to enable the S3 endpoint for the VPC.
-   */
-  enableS3Endpoint?: boolean;
-}
 
 export class PrerenderFargate extends Construct {
   readonly bucket: Bucket;
 
-    /**
-     * Constructs a new Prerender Fargate service.
-     * @param scope The scope of the construct.
-     * @param id The ID of the construct.
-     * @param props The properties of the Prerender Fargate service.
-     */
-    constructor(scope: Construct, id: string, props: PrerenderOptions) {
-        super(scope, id);
+  /**
+   * Constructs a new Prerender Fargate service.
+   * @param scope The scope of the construct.
+   * @param id The ID of the construct.
+   * @param props The properties of the Prerender Fargate service.
+   */
+  constructor(scope: Construct, id: string, props: PrerenderFargateOptions) {
+    super(scope, id);
+
+    const {
+      tokenUrlAssociation,
+      certificateArn,
+      vpcId,
+      maxInstanceCount,
+      instanceMemory,
+      instanceCPU,
+      expirationDays,
+      enableS3Endpoint,
+      enableRedirectCache,
+      desiredInstanceCount,
+      bucketName,
+      domainName,
+      prerenderName,
+    } = props;
```

Comment:
I feel like using `props.{blah}` makes it clearer to see where the variables are coming from. But maybe that's just me :man_shrugging: If no one else has an issue with this I don't mind :slightly_smiling_face: